calyx-ir/src/structure.rs

```diff
@@ -538,6 +538,15 @@ impl<T> Assignment<T> {
         }
         self.guard.for_each(&mut |port| f(&port).map(Guard::port))
     }
+
+    /// Reads through assignments unmutably
+    pub fn iter_ports(&self) -> impl Iterator<Item = RRC<Port>> {
+        self.guard
+            .all_ports()
+            .into_iter()
+            .chain(std::iter::once(Rc::clone(&self.dst)))
+            .chain(std::iter::once(Rc::clone(&self.src)))
+    }
```

Comment:
Is there no function that already does this? Also, the comment does not really explain what the method is supposed to do.